### PR TITLE
Remove Travis CI and add GitHub workflows CI

### PR DESCRIPTION
**Pull request by unfulvio-godaddy**
### Description of the Change

Replace Travis with a new CI workflow using GitHub actions. 
Having tests run again on each PR should give any contributor more confidence when opening a pull request. 

I had to bump min php 7.3 to avoid issues with composer. We should version composer.lock after this. 

Have also added an action that generates a coverage badge based on PHPUnit test coverage output. This is a static badge and not one dynamically generated with a third party service like Coveralls, CodeCov, etc. To use [Coveralls as a GitHub action](https://github.com/marketplace/actions/coveralls-github-action) looks like we would need additional setup since lcov is not a coverage format that phpunit supports. Having a static badge should be good enough though, in the future we could have it automatically updated to README.md as part of a release bump.

### How to test the Change

- [x] Tests should run on GitHub CI and pass.
- [x] Test coverage should be output in the same workflow
- [x] A badge will be appended and updated as GitHub comment with the current coverage %

This change is ignoring Behat at the moment. Need discussion to understand if we still need it or could limit test coverage to unit tests only. Behat could still be run by appending `vendor/bin/behat` right after `phpunit` in the corresponding action step.

Suggested list of follow up improvements:

- Commit composer.lock to GitHub (currently ignored)
- Set PHP standards to use and introduce PHPCS in workflow
- Perhaps later we can also add PHPStan

### Changelog Entry

- Require PHP 7.3 (from 7.1)
- Dropped travis.yml
- Added .github/workflows/ci.yml
- Tweaked phpunit.xml
- Adds `@covers` tags to tests (now a requirement)
- Updates .gitignore

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,19 +4,27 @@
...
-		"classmap": ["php/WP_Mock.php"]
+		"psr-4"   : {
+		  "WP_Mock\\": "./php/WP_Mock"
+		},
+		"classmap": [
+		  "php/WP_Mock.php"
+		]
+	},
+	"config": {
+		"platform-check": false
```
Comment (agibson-godaddy) [inline composer.json:27]: Why was this added?

**Review by agibson-godaddy — COMMENTED** [review]
Looks good, just left a question.

File: composer.json (hunk 1 above)
Comment (unfulvio-godaddy) [reply composer.json]: sorry what I really meant was this: https://github.com/10up/wp_mock/pull/167/commits/82d4ec65cf034a8ef6a628bf609619fb3cb0a023
thanks for reminding me

**Review by agibson-godaddy — DISMISSED** [review]
LGTM

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
+
+# IDEs
+.project/
+.settings/
+.idea/
+*.code-workspace
+*.sublime-project
+*.sublime-workspace
+*.iml
+.vscode/
```
Comment (rneudorf-godaddy) [inline .gitignore:15]: I would argue that the repo should not contain development environment specific ignores. Contributors can/should set their own local global ignores.

Comment (agibson-godaddy) [reply .gitignore:15]: I think you're definitely technically correct on that one. But with an open source repo that may have any number of contributors (including those that haven't configured their global ignore), maybe this just makes things easier?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    steps:
+      - uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2.4.2
+
+      - name: Switch to PHP ${{ env.PHP_VERSION }}
+        run: |
+          sudo update-alternatives --set php /usr/bin/php$PHP_VERSION
+          sudo update-alternatives --set phar /usr/bin/phar$PHP_VERSION
+          sudo update-alternatives --set phar.phar /usr/bin/phar.phar$PHP_VERSION
+          sudo update-alternatives --set phpize /usr/bin/phpize$PHP_VERSION
+          sudo update-alternatives --set php-config /usr/bin/php-config$PHP_VERSION
```
Comment (llessa-godaddy) [inline .github/workflows/ci.yml:34]: I think we could use https://github.com/marketplace/actions/setup-php-action to make this cleaner.

File: .gitignore (hunk 2 above)
Comment (unfulvio-godaddy) [reply .gitignore:15]: Yes my thinking was the same as @agibson-godaddy

File: .github/workflows/ci.yml (hunk 3 above)
Comment (unfulvio-godaddy) [reply .github/workflows/ci.yml:35]: as discusses with @llessa-godaddy we don't need to refactor that now, however we added a code coverage badge which may be handy

**Review by nmolham-godaddy — DISMISSED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        # Otherwise, current coverage can be viewed in the action output.
+        # Tests should be run randomly to catch any test dependency issue.
+        run: |
+          vendor/bin/phpunit --order-by="random" --coverage-cache="$COVERAGE_CACHE_PATH" --coverage-cobertura=./cobertura.xml --coverage-clover=./clover.xml --coverage-text
+
+      # TODO decide later if to keep using Coveralls
+      #- name: Upload coverage results to Coveralls
+      #  if: ${{ env.PHP_VERSION == '8.1' }}
+      #  run: |
+      #    vendor/bin/php-coveralls --coverage_clover="./clover.xml" --json_path="./coveralls-upload.json" -v
```
Comment (unfulvio-godaddy) [inline .github/workflows/ci.yml:79]: Coveralls won't work currently because it needs some envs set up - PHPUnit Code coverage and a badge output is currently working from this PR on the other hand

**Review by agibson-godaddy — APPROVED** [review]
